### PR TITLE
P-862 add chain-aio-image

### PR DESCRIPTION
**Pull request by m1iktea**
### Context

To resolve this [task](https://linear.app/litentry/issue/P-862/prepare-a-new-docker-image-litentry-parachain-for-testing)

### Labels

- C1

### How (Optional)

I've added a new image litentry/litentry-chain-aio, based on the build logic of litentry-parachain. In the Dockerfile, I've added a new target to differentiate it from the regular litentry-parachain image.

The AIO (all-in-one) image integrates relay-chain and parachain, as well as automated registration features.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -434,6 +436,13 @@ jobs:
           path: litentry-parachain-dev.tar.gz
           if-no-files-found: error
 
+      - name: Upload docker image
```
Comment (BillyWooo) [inline .github/workflows/ci.yml]: should not have the same name as previous step.

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+PARACHAIN_BASEDIR="/opt/litentry/parachain"
+REPO_DIR="/code/litentry-parachain"
+
+check(){
+    if [ -z "$CHAIN" ]; then
+        export CHAIN="rococo"
+    fi
+
+    ALLOWED_VALUES=("rococo" "litmus" "litentry")
```
Comment (BillyWooo) [inline docker/entrypoint.sh:14]: I think we only 'rococo' is enough.

File: docker/Dockerfile [hunk 3]
```diff
@@ -49,3 +49,59 @@ VOLUME ["/data"]
...
+    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" && \
+    [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion" && \
+    nvm install 18 && \
+    nvm use 18 && \
+    apt-get clean && \
+    rm -rf /var/cache/apt/lists 
+
+RUN useradd -m -u 1000 -U -s /bin/sh -d /litentry litentry && \
+	mkdir -p /opt/litentry/parachain /code/litentry-parachain && \
+	chown -R litentry:litentry /opt/litentry
```
Comment (BillyWooo) [inline docker/Dockerfile:88]: I think here miss to change the ownership of `/code/litentry-parachain`.

**Review by BillyWooo — COMMENTED** [review]
In general, it looks good 👍 .
I left some comments. 
Please fix the CI failure. Then we test it a bit.

File: docker/Dockerfile (hunk 3 above)
Comment (m1iktea) [reply docker/Dockerfile:88]: The ownership change of /code/litentry-parachain must occur after the COPY operation. There is already a line for this after line 95.

**Review by BillyWooo — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -422,18 +422,27 @@ jobs:
         if: needs.set-condition.outputs.rebuild_parachain == 'false'
         run: |
           docker pull litentry/litentry-parachain:latest
+          docker pull litentry/litentry-chain-aio:latest
 
       - name: Save docker image
         run: |
           docker save litentry/litentry-parachain:latest | gzip > litentry-parachain-dev.tar.gz
+          docker save litentry/litentry-chain-aio:latest | gzip > litentry-chain-aio-dev.tar.gz
```
Comment (Kailai-Wang) [inline .github/workflows/ci.yml]: You could save multiple images to one tgz file (if you want), so that we once need upload/download once 😂

**Review by Kailai-Wang — APPROVED** [review]
Not much to complain, thank you!

File: .github/workflows/ci.yml (hunk 4 above)
Comment (m1iktea) [reply .github/workflows/ci.yml]: ok

**Pull request merged**